api/lib/routes/metricHelper.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+exports.parseParameter = function(req) {
+  var routeHelper = require("./routeHelper");
+  var parameters = {};
+  var appId = req.params.app_id;
+  var metricType = req.query["metric-type"];
+  var startTime = req.query["start-time"];
+  var endTime = req.query["end-time"];
+  var order = req.query["order"];
+  var page = req.query["page"];
+  var resultsPerPage = req.query["results-per-page"];
+  var ASC = routeHelper.ASC;
+  var DESC = routeHelper.DESC;
+  if (!appId) {
+    return { valid: false, message: "app_id is required" };
+  } else {
+    parameters.appId = appId;
+  }
+  if (routeHelper.isMissing(metricType)) {
+    return { valid: false, message: "metric-type is required" };
+  } else {
+    if (!routeHelper.isString(metricType)) {
+      return { valid: false, message: "metric-type must be a string" };
+    }else{
+      parameters.metricType = metricType;
+    }
+  }
+  if (routeHelper.isMissing(startTime)) {
+    parameters.startTime = 0;
+  } else {
+    startTime = parseInt(startTime);
+    if (!routeHelper.isInteger(startTime)) {
+      return { valid: false, message: "start-time must be an integer" };
+    } else {
+      parameters.startTime = startTime;
+    }
+  }
+
+  if (routeHelper.isMissing(endTime)) {
+    parameters.endTime = -1;
+  } else {
+    endTime = parseInt(endTime);
+    if (!routeHelper.isInteger(endTime)) {
+      return { valid: false, message: "end-time must be an integer" };
+    } else {
+      parameters.endTime = endTime;
+    }
+  }
+  if (routeHelper.isMissing(order)) {
+    parameters.order = DESC;
+  } else {
+    if (!routeHelper.isString(order)) {
+      return { valid: false, message: "order must be a string" };
+    } else {
+      order = order.toLowerCase();
+      if (order !== DESC && order !== ASC) {
+        return { valid: false, message: "order must be desc or asc" };
+      } else {
+        parameters.order = order;
+      }
+    }
+  }
+  if (routeHelper.isMissing(page)) {
+    parameters.page = 1;
+  } else {
+    page = parseInt(page);
+    if (!routeHelper.isInteger(page)) {
+      return { valid: false, message: "page must be an integer" };
+    }
+    if (page <= 0) {
+      return { valid: false, message: "page must be greater than 0" };
+    }
+    parameters.page = page;
+  }
+
+  if (routeHelper.isMissing(resultsPerPage)) {
```

Comment:
@boyang9527  for scaling history, we set a default value 10 if resultPerpage is missing, so here you mean we change the rule just for metrics or apply it to scaling history, too?